src/azul/portal_service.py

```diff
@@ -197,7 +201,16 @@ def _read_db(self, version: str) -> JSONs:
                                               Key=self.object_key,
                                               VersionId=version)
         except self.client.exceptions.NoSuchKey:
+            # We hypothesis that NoSuchKey is raised when the object has been
+            # created but has not materializied yet.
             raise NoSuchObjectVersion(version)
+        except ClientError as e:
+            # We hyposthesis that NoSuchVersion is raised when the object has
+            # been overwritten but the overwrite has not materilaized yet.
+            if 'NoSuchVersion' in e.response['Error']['Code']:
```

Comment:
Why the substring search? The AWS documentation clearly states that the error code *is* `NoSuchVersion`.

Reply:
I was originally inspecting the exception message (An error occurred (NoSuchVersion) when calling the GetObject operation: The specified version does not exist.) and not the response. I later changed to inspect the e.response, but had left the inspection over.

Changed to string comparison.